src/main/java/com/hms_networks/sc/cumulocity/data/CConnectorDataMgr.java

```diff
@@ -0,0 +1,506 @@
+import com.hms_networks.sc.cumulocity.api.CConnectorMqttMgr;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Class for managing data from the Ewon's historical data queue corresponding systems.
+ *
+ * @since 1.0.0
+ * @author HMS Networks, MU Americas Solution Center
+ */
+public class CConnectorDataMgr {
+
+  /** The size of the array with expected tag name components. */
+  private static final int SPLIT_TAG_NAME_ARRAY_SIZE = 3;
+
+  /** The delimiter used to split the tag name into components. */
+  public static final String SPLIT_TAG_NAME_DELIMITER = "/";
+
+  /** The default Cumulocity data series name if one is not included in the tag name. */
+  private static final String DEFAULT_SERIES_VALUE = "0";
+
+  /**
+   * The default interval (in milliseconds) to poll the historical data queue in the event it cannot
+   * be read from the configuration file.
+   */
+  public static final long QUEUE_DATA_POLL_INTERVAL_MILLIS_DEFAULT = 10000;
+
+  /** The minimum memory (in bytes) required to perform a poll of the data queue. */
+  public static final int QUEUE_DATA_POLL_MIN_MEMORY_BYTES = 5000000;
+
+  /** The time (in milliseconds) that the data queue must be behind by before warning the user. */
+  public static final long QUEUE_DATA_POLL_BEHIND_MILLIS_WARN = 300000;
+
+  /** The maximum number of historical data queue poll failures before a reset is triggered. */
+  public static final int QUEUE_DATA_POLL_FAILURE_RESET_THRESHOLD = 5;
+
+  /**
+   * The name of the diagnostic tag that is populated with the number of seconds that the queue is
+   * running behind.
+   */
+  public static final String QUEUE_DIAGNOSTIC_TAG_RUNNING_BEHIND_SECONDS_NAME =
+      "ConnectorQueueBehindSeconds";
+
+  /**
+   * The description of the diagnostic tag that is populated with the number of seconds that the
+   * queue is running behind.
+   */
+  public static final String QUEUE_DIAGNOSTIC_TAG_RUNNING_BEHIND_SECONDS_DESC =
+      "Diagnostic tag containing the amount of time, in seconds, that the connector data queue is"
+          + " running behind.";
+
+  /**
+   * The type of the diagnostic tag that is populated with the number of seconds that the queue is
+   * running behind.
+   */
+  public static final int QUEUE_DIAGNOSTIC_TAG_RUNNING_BEHIND_SECONDS_TYPE = 2;
+
+  /**
+   * The name of the diagnostic tag that is monitored for a request to forcibly reset the queue time
+   * tracker.
+   */
+  public static final String QUEUE_DIAGNOSTIC_TAG_FORCE_RESET_NAME = "ConnectorQueueForceReset";
+
+  /**
+   * The description of the diagnostic tag that is monitored for a request to forcibly reset the
+   * queue time tracker.
+   */
+  public static final String QUEUE_DIAGNOSTIC_TAG_FORCE_RESET_DESC =
+      "Diagnostic tag which can be used to request the connector to reset the queue time tracker.";
```

Comment:
These descriptions were approved as part of a PR for the TW connector, and thus should remain consistent across connectors. Ref: https://github.com/hms-networks/flexy-thingworx-connector/blob/main/src/main/java/com/hms_networks/americas/sc/thingworx/TWConnectorConsts.java#L52-L109

The idea is that this will end up in the library when time permits and won't be specific to any connector.